.envrc

```diff
@@ -0,0 +1,27 @@
+# This file is used by direnv to setup the environment when entering to use fvm's flutter.
+
+# Comment out the next line if you want to use your system flutter.
+PATH_add .fvm/flutter_sdk/bin
+
+# Check flutter version matches what's in .fvmrc
+check_flutter_version() {
+  local fvm_flutter_version_string=$(cat .fvmrc)
+  # Fetch string from format `"flutter": "18340ea16c"``
+  local fvm_flutter_version=$(echo "$fvm_flutter_version_string" | grep -o '"flutter": "[a-z0-9]*"' | cut -d '"' -f 4)
```

Comment:
The behavior of `cut` is finicky to reason through. Since there's already a regexp describing what the line should look like, better to take that just slightly further to identify which part of the line we want:
```suggestion
  local fvm_flutter_version=$(
    <.fvmrc perl -lne 'print $1 if (/"flutter": "([a-z0-9]*)"/)'
  )
```

Then two other points:
* should be a-f, not a-z :-) (and let's write it as `[0-9a-f]`, so the digits are in order of their value)
* a fine point of shell semantics which shellcheck will tell you about — more about that in my next comment

Reply:
(Oh, and since the file being parsed here is actually JSON, the cleanest solution would be to use `jq` instead. But because this is a script that will run for all developers in this repo, best to do without that extra dependency. We do regularly use `jq` in scripts that only a smaller set of maintainers are expected to need to run.)